### PR TITLE
Include icon for xdg menu entry

### PR DESCRIPTION
**Pull request by tobtoht**
Closes #2292

### CODE REVIEW

**Comment by selsta** [comment]
Does this get written every time the GUI gets opened? Is this ok? Looks good otherwise.

**Comment by tobtoht** [comment]
>Does this get written every time the GUI gets opened?

Hm, yeah. It's the same behavior as the .desktop, but it's not really necessary for the icon. I can have it check for existence.

**Comment by selsta** [comment]
@xiphon Is writing on every start ok? Would you prefer existence check?

File: src/qt/utils.cpp [hunk 1]
```diff
@@ -100,7 +114,6 @@ void registerXdgMime(QApplication &app){
 
     if (TailsOS::detect() && TailsOS::detectDotPersistence() && TailsOS::usePersistence) {
         TailsOS::persistXdgMime(filePath, mime);
-        return;
```
Comment (xiphon) [inline src/qt/utils.cpp]: ?

File: qml.qrc [hunk 2]
```diff
@@ -241,5 +241,6 @@
         <file>images/edit.svg</file>
         <file>images/arrow-right-in-circle-outline-medium-white.svg</file>
         <file>images/tails-grey.png</file>
+        <file>images/appicons/64x64.png</file>
```
Comment (xiphon) [inline qml.qrc]: Could you please move this before the line 239 ` <file>images/check-white.svg</file>`

File: src/qt/utils.cpp [hunk 3]
```diff
@@ -58,6 +59,19 @@ bool fileWrite(QString path, QString data) {
     return false;
 }
 
+bool pixmapWrite(QString path, QPixmap pixmap) {
```
Comment (xiphon) [inline src/qt/utils.cpp]: Take `path` by const ref or std::move it, take `pixmap` by const ref.

**Comment by xiphon** [comment]
> @xiphon Is writing on every start ok? Would you prefer existence check?

I don't like malware-like behavior, i.e. not prompting a user and silently dropping some files to the system at every startup.
And there is no way to cleanup the system afterwards.

But the code doing this is already there.
Would say it's out of scope of the current PR.

Would be nice to have existence check at least.

File: src/qt/utils.cpp (hunk 1 above)
Comment (tobtoht) [reply src/qt/utils.cpp]: desktop file and icon should also be written to default locations on Tails.

**Comment by tobtoht** [comment]
Replaced by #2984

**Pull request closed without merging**